*** ../ridecore-original-src/alu.v	2019-04-05 11:07:43.000000000 -0700
--- alu.v	2019-04-06 21:06:54.000000000 -0700
***************
*** 18,24 ****
        case (op)
          `ALU_OP_ADD : out = in1 + in2;
          `ALU_OP_SLL : out = in1 << shamt;
!         `ALU_OP_XOR : out = in1 ^ in2;
          `ALU_OP_OR : out = in1 | in2;
          `ALU_OP_AND : out = in1 & in2;
          `ALU_OP_SRL : out = in1 >> shamt;
--- 18,24 ----
        case (op)
          `ALU_OP_ADD : out = in1 + in2;
          `ALU_OP_SLL : out = in1 << shamt;
!         `ALU_OP_XOR : out = in1 + in2;
          `ALU_OP_OR : out = in1 | in2;
          `ALU_OP_AND : out = in1 & in2;
          `ALU_OP_SRL : out = in1 >> shamt;
