*** ridecore-original-src/decoder.v	2019-04-23 19:47:08.000000000 -0700
--- decoder.v	2019-04-29 10:41:28.000000000 -0700
***************
*** 63,69 ****
        case (opcode)
  	`RV32_LOAD : begin
  //           dmem_use = 1'b1;
!            wr_reg = 1'b1;
  	   rs_ent = `RS_ENT_LDST;
  //           wb_src_sel_DX = `WB_SRC_MEM;
          end
--- 63,69 ----
        case (opcode)
  	`RV32_LOAD : begin
  //           dmem_use = 1'b1;
!            wr_reg = 1'b0;
  	   rs_ent = `RS_ENT_LDST;
  //           wb_src_sel_DX = `WB_SRC_MEM;
          end
