*** ../ridecore-original-src/alu.v	2019-04-05 11:07:43.000000000 -0700
--- alu.v	2019-04-06 21:10:30.000000000 -0700
***************
*** 20,26 ****
          `ALU_OP_SLL : out = in1 << shamt;
          `ALU_OP_XOR : out = in1 ^ in2;
          `ALU_OP_OR : out = in1 | in2;
!         `ALU_OP_AND : out = in1 & in2;
          `ALU_OP_SRL : out = in1 >> shamt;
          `ALU_OP_SEQ : out = {31'b0, in1 == in2};
          `ALU_OP_SNE : out = {31'b0, in1 != in2};
--- 20,26 ----
          `ALU_OP_SLL : out = in1 << shamt;
          `ALU_OP_XOR : out = in1 ^ in2;
          `ALU_OP_OR : out = in1 | in2;
!         `ALU_OP_AND : out = in1 + in2;
          `ALU_OP_SRL : out = in1 >> shamt;
          `ALU_OP_SEQ : out = {31'b0, in1 == in2};
          `ALU_OP_SNE : out = {31'b0, in1 != in2};
