*** ./ridecore-original-src/decoder.v	2019-04-29 14:59:52.000000000 -0700
--- decoder.v	2019-05-06 07:29:11.000000000 -0700
***************
*** 171,177 ****
  	 */
          `RV32_AUIPC : begin
             uses_rs1 = 1'b0;
!            src_a_sel = `SRC_A_PC;
             imm_type = `IMM_U;
             wr_reg = 1'b1;
          end
--- 171,177 ----
  	 */
          `RV32_AUIPC : begin
             uses_rs1 = 1'b0;
!            src_a_sel = `SRC_A_ZERO;
             imm_type = `IMM_U;
             wr_reg = 1'b1;
          end
