*** ridecore-original-src/decoder.v	2019-06-17 00:30:35.000000000 -0700
--- decoder.v	2019-06-17 00:35:19.000000000 -0700
***************
*** 172,178 ****
          `RV32_AUIPC : begin
             uses_rs1 = 1'b0;
             src_a_sel = `SRC_A_PC;
!            imm_type = `IMM_U;
             wr_reg = 1'b1;
          end
          `RV32_LUI : begin
--- 172,178 ----
          `RV32_AUIPC : begin
             uses_rs1 = 1'b0;
             src_a_sel = `SRC_A_PC;
!            imm_type = `IMM_J;
             wr_reg = 1'b1;
          end
          `RV32_LUI : begin
